week-1/Homework/mandatory/5-magic-8-ball.js

```diff
@@ -1,61 +1,82 @@
-/**
+//I couldnt write this code alone. we did together with Ferhat, Serkan and Tayfun
 
-Let's peer into the future using a Magic 8 Ball!
-https://en.wikipedia.org/wiki/Magic_8-Ball 
-
-There are a few steps to being able view the future though:
-* Ask a question
-* Shake the ball
-* Get an answer
-* Decide if it's positive or negative
-
-The question can be anything, but the answers are fixed,
-and have different levels of positivity or negativity.
-
-Below are the possible answers:
-
-## Very positive
-It is certain.
-It is decidedly so.
-Without a doubt.
-Yes - definitely.
-You may rely on it.
-
-## Positive
-As I see it, yes.
-Most likely.
-Outlook good.
-Yes.
-Signs point to yes.
-
-## Negative
-Reply hazy, try again.
-Ask again later.
-Better not tell you now.
-Cannot predict now.
-Concentrate and ask again.
-
-## Very negative
-Don't count on it.
-My reply is no.
-My sources say no.
-Outlook not so good.
-Very doubtful.
-*/
-
-// This should log "The ball has shaken!"
 // and return the answer.
-function shakeBall() {}
-
+//I am defining all possible answers into ane answers array
+const answers = [
+  "It is certain",
+  "It is decidedly so",
+  "Without a doubt",
+  "Yes - definitely",
+  "You may rely on it",
+  "As I see it, yes",
+  "Most likely",
+  "Outlook good",
+  "Yes",
+  "Signs point to yes",
+  "Reply hazy, try again",
+  "Ask again later",
+  "Better not tell you now",
+  "Cannot predict now",
+  "Concentrate and ask again",
+  "Don't count on it",
+  "My reply is no",
+  "My sources say no",
+  "Outlook not so good",
+  "Very doubtful"
+];
+//Then create different arrays according to their levels
+const veryPositive = [
+  "It is certain",
+  "It is decidedly so",
+  "Without a doubt",
+  "Yes - definitely",
+  "You may rely on it"
+];
+const positive = [
+  "As I see it, yes",
+  "Most likely",
+  "Outlook good",
+  "Yes",
+  "Signs point to yes"
+];
+const negative = [
+  "Reply hazy, try again",
+  "Ask again later",
+  "Better not tell you now",
+  "Cannot predict now",
+  "Concentrate and ask again"
+];
+const veryNegative = [
+  "Don't count on it",
+  "My reply is no",
+  "My sources say no",
+  "Outlook not so good",
+  "Very doubtful"
+];
+function shakeBall() {
+  console.log("The ball has shaken!");
+  let i = Math.floor(Math.random() * answers.length);
+  return answers[i];
+}
 // The answer should come from shaking the ball
-let answer;
-
+const result = shakeBall();
 // When checking the answer, we should tell someone if the answer is
 // - very positive
 // - positive
 // - negative
 // - very negative
-function checkAnswer() {}
+function checkAnswer() {
+  if (veryPositive.includes(result)) {
+    return "very positive";
+  } else if (positive.includes(result)) {
+    return "positive";
+  } else if (negative.includes(result)) {
+    return "negative";
+  } else {
+    return "very negative";
+  }
+}
+
```

Comment:
Very good solution, you saved so much code by doing it this way!

One small detail, given how `checkAnswer()` is used in the test below, I think it was supposed to accept an `answer` parameter (eg `function checkAnswer(answer) { ... }`) instead of reading the value in the outer variable `result`.